src/dotnet/CarbonAware.WebApi/api.yaml

```diff
@@ -0,0 +1,152 @@
+openapi: 3.0.0
+info:
+  description: 'Carbon Aware REST API'
+  version: 0.1.0
+  title: Carbon Aware REST API
+
+paths:
+  '/sci-scores':
+    post:
+      summary: Calculate the SCI score using the configured data sources
+      description: ''
+      operationId: calculateSciScore
+      requestBody:
+        '$ref': '#/components/requestBodies/MarginalCarbonIntensityCalculation'
+        examples: 
+          sci-score-calculation:
+            '$ref': '#/components/examples/sci-score-calc-example'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                '$ref': '#/components/schemas/SciScore'
+              examples: 
+                sci-score-response:
+                  '$ref': '#/components/examples/sci-score-response-example'
+        '400':
+          description: bad request
+          content:
+            application/json:
+              schema:
+                '$ref': '#/components/schemas/CarbonAwareError'
+  '/sci-scores/marginal-carbon-intensity':
+    post:
+      summary: Calculate the average marginal carbon intensity for a given location and time duration.
+      description: ''
+      operationId: calculateMarginalCarbonIntensity
```

Comment:
Seems like there would be a lot of dup code to some degree on the aggregators given the fact that now there is an average already implemented but not consumed by upper layer.

Reply:
This spec is meant to support the reporting use-case.  The amount of duplicate code would be up to the implementor.  There are many approaches one could take:

- Duplicate the code, and allow both implementations to evolve independently.
- Remove the implementation that is not being consumed by an upper layer in favor of one that is.
- Have this endpoint leverage the existing aggregator.

All trade-offs worthy of consideration.

Reply:
Understood. I'm open too for evolving the impl to show that it is possible to have different ways of doing it as a mechanism to illustrate the flexibility.